### PR TITLE
docs: :memo: add concepts page

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Addresses: 
[AB#36137](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/36137)
[AB#36142](https://statoil-proview.visualstudio.com/9949ad5e-7d15-4531-901e-296574079ee1/_workitems/edit/36142)

image

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] Documentation and comments is added where needed.
-   [ ] My code is covered by tests.
-   [ ] UX has reviewed relevant UI contributions.

## Definition of Done
- [x] PR title and description are to the point and understandable
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by Noggling — APPROVED** [review]

File: documentation/src/content/docs/concepts/boundaries.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: 'Boundaries'
+description: 'A section about suspense and error boundaries'
+---
+
+## Suspsense and error boundaries
+
+Boundaries are heavily used throughout the workspace package.
+Every custom component passed from the config will be wrapped with a suspense and error boundary.
```
Comment (jaysencpp) [inline documentation/src/content/docs/concepts/boundaries.mdx:9]: What is "custom component"? A react component? A custom table cell in the table config? Will default components (Ag grid?) not be wrapped with suspense and error boundary?

File: documentation/src/content/docs/concepts/boundaries.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+---
+
+## Suspsense and error boundaries
+
+Boundaries are heavily used throughout the workspace package.
+Every custom component passed from the config will be wrapped with a suspense and error boundary.
+
+### Suspense
+
+If a [promise](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/promise) is thrown the suspense boundary will render a fallback component while the promise is in the state `pending`.
```
Comment (jaysencpp) [inline documentation/src/content/docs/concepts/boundaries.mdx:13]: `If a [promise](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/promise) is thrown`
Do promises throw or reject?

Comment (jaysencpp) [reply documentation/src/content/docs/concepts/boundaries.mdx:13]: Actually I don't understand the full sentence. If you `throw` a `Promise` a suspense boundary will render a fallback component?

File: documentation/src/content/docs/concepts/boundaries.mdx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+For most of the cases our fallback component will be the `CircularProgess` from [EDS](https://eds.equinor.com).
+
+### Error boundary
+
+React supports error boundaries but do not provide any native component for using it.
+For our implementation we chose the package [react-error-boundary](https://www.npmjs.com/package/react-error-boundary).
+Just as with suspense an error boundary is triggered using a thrown promise in a component. The difference is the error boundary triggers when the promise changes its state to `rejected`.
```
Comment (jaysencpp) [inline documentation/src/content/docs/concepts/boundaries.mdx:20]: `triggered using a thrown promise` is thrown promise a thing or?

File: documentation/src/content/docs/concepts/boundaries.mdx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+### Taking control
+
+If the default suspense and error boundaries does not satisfy your needs, wrap your components in your own suspense and error boundary.
```
Comment (jaysencpp) [inline documentation/src/content/docs/concepts/boundaries.mdx:24]: Do you have an example on how to do this? Any `best practice` would be good to add, either here or in another section with examples

Comment (jaysencpp) [inline documentation/src/content/docs/concepts/boundaries.mdx:24]:
```suggestion
If the default suspense and error boundaries do not satisfy your needs, wrap your components in your own suspense and error boundary.

```

File: documentation/src/content/docs/concepts/boundaries.mdx (hunk 2 above)
Comment (Gustav-Eikaas) [reply documentation/src/content/docs/concepts/boundaries.mdx:13]: To trigger a suspense you throw new Promise(). But a promise itself rejects it doesnt throw

File: documentation/src/content/docs/concepts/boundaries.mdx (hunk 3 above)
Comment (Gustav-Eikaas) [reply documentation/src/content/docs/concepts/boundaries.mdx:20]: Yes I can elaborate on that section, right now it's an experimental feature from React. Ill add a link to the beta docs atleast

File: documentation/src/content/docs/concepts/boundaries.mdx (hunk 4 above)
Comment (Gustav-Eikaas) [reply documentation/src/content/docs/concepts/boundaries.mdx:24]: Yes will add example, so much text and no examples in this section

File: documentation/src/content/docs/concepts/boundaries.mdx (hunk 1 above)
Comment (Gustav-Eikaas) [reply documentation/src/content/docs/concepts/boundaries.mdx:9]: Hmm probably not, I should add that

**Pull request merged**